pkg/es/client/index_client.go

```diff
@@ -244,12 +244,10 @@ func (i IndicesClient) version() (uint, error) {
 // CreateTemplate an ES index template
 func (i IndicesClient) CreateTemplate(template, name string) error {
 	endpointFmt := "_template/%s"
-	if v, err := i.version(); err == nil {
-		if v >= 8 {
-			endpointFmt = "_index_template/%s"
-		}
-	} else {
+	if v, err := i.version(); err != nil {
 		return err
+	} else if v >= 8 {
```

Comment:
so linter does not complain about this one? If so, why did we have to change many other places to move variables out of `if` scope to the previous scope, e.g. in cmd/remote-storage/app/flags.go?

Reply:
Or does the linter only allow this when there is another condition `else if`, but with plain `else` it would complain that it's not necessary? (which is not true due to variable scope)

Reply:
The latter. It only complains when there is an `else`  
The suggestion by the linter is this:
```
indent-error-flow: if block ends with a return statement, so drop this else and outdent its block (move short variable declaration to its own line if necessary)
```
I think its not checking for any complex logic but just for the basic `if else` case.

I can move the declaration variable declaration out of the `if` scope and convert the `else if` into a plain `if` out of the `if` scope for the above case.